src/http/server/response.cr

```diff
@@ -50,6 +50,7 @@ class HTTP::Server
     def reset
       # This method is called by RequestProcessor to avoid allocating a new instance for each iteration.
       @headers.clear
+      @headers.sent = false
```

Comment:
Shouldn't this happen in `Headers#clear`?

Reply:
That was my first thought, but I'm not sure. It's not obvious that any time you want to clear the headers you also want to mark them as not sent.

I think that `clear` method delegates to Hash, so effectively jumping over the `sent?` guard. I think `HTTP::Headers` should stop forwarding stuff to Hash. Then we need to move `sent = false` above this line.

This looks like it will take a long time to settle, so for now I'm abandoning this PR. Anyone feel free to continue this.

Reply:
Clear already changes the headers, so it can't work any other way than with `sent == false`. It either needs to call `check_not_sent` or implicitly set `sent = false`. When doing the latter, we can also revert from `sent=` to `mark_as_send!` and the only way to remove the sent flag is calling `clear`. That should actually cover all use cases.
Maybe `clear` shoud be renamed to `reset` to better communicate its purpose.

Reply:
The thing is, imagine the headers were already sent. Someone calls `clear`, appends a few headers, everything works fine. Except that it has no effect on the output because the headers were sent and you didn't get a runtime error.

This is what I'm trying to avoid.

Reply:
Then `clear` shouldn't be part of the public API or make it `clear!`/`reset!` to emphasize the possible dangers related to it.